### ISSUE TITLE
Error when trying to deploy on heroku : no module named requests

### ISSUE THREAD

**Issue by tutacrypto** [lnpay/lnpay-py#1]
Hello, 

I got an `error : no module named requests`, during the installation of `lnpay-py` when trying to deploy an app on Heroku. 
The `requests` module is in my `requirements.txt` before `lnpay-py`. 

I contacted Heroku support, and here is what they told me : 

"Hi Gustave,

Sorry about that! I completely missed it in your first comment, thank you for clarifying.

Our Python language owner was kind enough to take a look at your situation and I've included the feedback he provided below. Ultimately most likely issue here is that the `lnpay-py` package is broken.

Python packages can be shipped as either wheels (newer; preferred) or sdists (source distributions). For sdists, when the package is installed, setuptools invokes `setup.py` as part of the install.

Looking at the package it seems `setup.py` [imports a file](https://github.com/lnpay/lnpay-py/blob/757f4ba955636bb09514fa7ced80bab495563fd3/setup.py#L4) that [in turn](https://github.com/lnpay/lnpay-py/blob/757f4ba955636bb09514fa7ced80bab495563fd3/lnpay_py/__init__.py#L1-L3) imports requests, meaning that the package actually has not only a run time dependency on requests, but also a build time dependency.

Adding `requests` to `requirements.txt` won't help, since during the very first pip install, requests won't yet have been installed when `lnpay-py`'s setup.py is being invoked. The Python ecosystem is partway through devising a way to handle build time dependencies (https://www.python.org/dev/peps/pep-0518/) though in this case the package is just accidentally importing it and doesn't necessarily need it.

To ultimately resolve the issue you'll need to reach out to the the author of the package to see if they can fix the import issue. You can specifically mention that the package maintainer should rework the `__version__` import such that it doesn't also import `requests` during pip install, since the current implementation will break installing into any clean environment, not just Heroku. In addition, the package maintainer should ideally also ship wheels, since they are the preferred package format (and would have wallpapered over this issue too):https://pythonwheels.com/

Does all of that make sense? Please let me know if you have any questions or if I can assist any further!

Thanks,
Lucas" 

I don't understand everything, but I hope you and others will! Don't hesitate to tell me if you want me to ask them more information.

**Comment by gustmtl** [comment]
I'm using lnpay-py as a library in my python app's server (using Flask framework).
Locally everything works fine, it's only when deploying to Heroku cloud provider that I'm having this issue.

**Comment by fiatjaf** [comment]
What if the error is in this line? https://github.com/lnpay/lnpay-py/blob/757f4ba955636bb09514fa7ced80bab495563fd3/setup.py#L4

Maybe a package shouldn't depend on itself at build time? I'm not sure.

**Comment by AnthonyRonning** [comment]
Yeah that's interesting, I can definitely replace that and see what happens, did that out of habit but I don't think I'm using it elsewhere anyways. I'll try to set up an example heroku project importing this library and play around with it over the next few days.

**Comment by Inspector-Butters** [comment]
this is still the case.
I usually have to run a `pip install requests` before installing lnpay and that works.

but its a pain when deploying to different platforms
